packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Utilities/FormattableStringHelpers.cs

```diff
@@ -97,5 +98,69 @@ public static string ReplaceLast(this string text, string oldValue, string newVa
             var position = text.LastIndexOf(oldValue, StringComparison.Ordinal);
             return position < 0 ? text : text.Substring(0, position) + newValue + text.Substring(position + oldValue.Length);
         }
+
+        internal static IReadOnlyList<FormattableString> BreakLines(FormattableString fs)
+        {
+            StringBuilder formatBuilder = new StringBuilder();
+            var args = new List<object?>();
+            List<FormattableString> result = new List<FormattableString>();
+            foreach ((var span, bool isLiteral, int index) in StringExtensions.GetPathParts(fs.Format))
+            {
+                // if isLiteral - put in formatBuilder
+                if (isLiteral)
+                {
+                    var splitNewLine = span.ToString().Split("\n");
+                    for (int i = 0; i < splitNewLine.Length; i++)
+                    {
+                        var part = splitNewLine[i];
+                        formatBuilder.Append(part);
+                        if (i < splitNewLine.Length - 1)
+                        {
+                            FormattableString formattableString = FormattableStringFactory.Create(formatBuilder.ToString(), args.ToArray());
+                            result.Add(formattableString);
+                            formatBuilder.Clear();
+                            args.Clear();
+                        }
+                    }
+                }
+                // if not Literal, is Args - recurse through Args and check if args has breaklines
+                else
+                {
+                    var arg = fs.GetArgument(index);
+                    if (arg is string str)
+                    {
+                        var splitNewLine = str.Split("\n");
```

Comment:
Need to update this arg check to recursively go through args and check for args within args.